src/components/DropdownButton.tsx

```diff
@@ -0,0 +1,42 @@
+import React, { useState, useRef } from "react";
+import { useOnClickOutside } from "usehooks-ts";
+
+interface DropdownButtonProps {
+  buttonText: string;
+  children: React.ReactNode;
+}
+
+export const DropdownButton = (props: DropdownButtonProps) => {
+  const [isActive, setActive] = useState(false);
+  const toggleActive = () => setActive(!isActive);
```

Comment:
Creo que en estos casos es buena práctica usar la función en lugar del valor:

```suggestion
  const toggleActive = () => setActive(x => !x);
```